keps/sig-node/3983-drop-in-configuration/README.md

```diff
@@ -0,0 +1,372 @@
+
+## Design Details
+
+
+### Test Plan
+
+[X] I/we understand the owners of the involved components may require updates to existing tests to make this code solid enough prior to committing the changes necessary to implement this enhancement.
+
+
+##### Prerequisite testing updates
+
+
+##### Unit tests
+
+* Unit tests will be added, details to be added here
+
+##### Integration tests
+
+* :
+
+##### e2e tests
+
+* :
+
+### Graduation Criteria
+
+#### Alpha
+
+Add ability to support drop-in configuration directory.
+
+#### Beta
+
+Add ability to easily view the effective configuration that is being used by kubelet. Details to be added during beta phase.
+
+### Upgrade / Downgrade Strategy
+
+Upgrades and downgrades are safe as far as Kubelet stability is concerned. It’s possible a vendor may ship vital pieces of configuration within a drop-in directory. If the Kubelet downgrades to a version that doesn’t support reading the drop-in directory, the kubelet will not recognize the "--config-dir" flag and risk failing. However, assuming that vendor left that the original `/etc/kubelet/kubelet.conf` is in a valid state, and the flag isn't specified, there should be no risk to the system. Any configuration that exists in a drop-in dir won't be applied, but that would not affect kubelet stability.
+
+
+### Version Skew Strategy
+
+All behavior change is encapsulated within the Kubelet, so there is no version skew possible within core Kubernetes. It is possible third party tools may attempt to utilize the Kubelet’s drop-in directory before the Kubelet is upgraded to support it, which would cause silent failures. It is the responsibility of these third party tools to ensure the Kubelet is new enough to support this.
+
+
+## Production Readiness Review Questionnaire
+
+
+### Feature Enablement and Rollback
+
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [ ] Feature gate
+  - Feature gate name: KubeletDropInConfig
+  - Components depending on the feature gate: Kubelet
+
+Aside from the featuregate, admins will also have to explicitly enable the kubelet flag to enable this.
+
+
+###### Does enabling the feature change any default behavior?
+
+No, upgrading to a version of the Kubelet with this feature will not enable the Kubelet to be configured with the drop-in directory if no flag is specified.
+
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Roll back and remove the `--config-dir` flag from the kubelet's CLI.
+
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+This feature will be re-enabled via adding back the `--config-dir` flag to the CLI, as mentioned above.
+
+
+###### Are there any tests for feature enablement/disablement?
+
+A test will be added to assemble a single, functional kubelet configuration object from various individual drop-in config files.
```

Comment:
there should be a test that the flag gives an error if the feature gate is not enabled